text/0000-language-semver.md

```diff
@@ -0,0 +1,494 @@
+ease the transition wherever possible.
+
+Known areas where change is expected include the following:
+
+- Destructors semantics:
+  - We plan to stop zeroing data and instead use marker flags on the stack,
+    as specified in [RFC 320]. This may affect destructors that rely on ovewriting
+    memory or using the `unsafe_no_drop_flag` attribute.
+  - Currently, panicing in a destructor can cause unintentional memory
+    leaks and other poor behavior (see [#14875], [#16135]). We are
+    likely to make panic in a destructor simply abort, but the precise
+    mechanism is not yet decided.
+  - Order of dtor execution within a data structure is somewhat
+    inconsistent (see [#744]).
+- The legal aliasing rules between unsafe pointers is not fully settled (see [#19733]).
+- The interplay of assoc types and lifetimes is not fully settled and can lead
+  to unsoundness in some cases (see [#23442]).
+- The trait selection algorithm is expected to be improved and made more complete over time.
+  It is possible that this will affect existing code.
+- [Overflow semantics][RFC 560]: in particular, we may have missed some cases.
+- Memory allocation in unsafe code is currently unstable. We expect to
+  be defining safe interfaces as part of the work on supporting
+  tracing garbage collectors (see [#415]).
+- The treatment of hygiene in macros is uneven (see [#22462], [#24278]). In some cases,
+  changes here may be backwards compatible, or may be more appropriate only with explicit opt-in
+  (or perhaps an alternate macro system altogether).
+- Lints will evolve over time (both the lints that are enabled and the
+  precise cases that lints catch). We expect to introduce a
+  [means to limit the effect of these changes on dependencies][#1029].
+- Stack overflow is currently detected via a segmented stack check
+  prologue and results in an abort. We expect to experiment with a
+  system based on guard pages in the future.
+- We currently abort the process on OOM conditions (exceeding the heap space, overflowing
+  the stack). We may attempt to panic in such cases instead if possible.
+- Some details of type inference may change. For example, we expect to
+  implement the fallback mechanism described in [RFC 213], and we may
+  wish to make minor changes to accommodate overloaded integer
+  literals. In some cases, type inferences changes may be better
+  handled via explicit opt-in.
+
+There are other kinds of changes that can be made in a minor version
+that may break unsafe code but which are not considered breaking
+changes, because the unsafe code is relying on things known to be
+intentionally unspecified. One obvious example is the layout of data
+structures, which is considered undefined unless they have a
+`#[repr(C)]` attribute.
+
+Although it is not directly covered by this RFC, it's worth noting in
+passing that some of the CLI flags to the compiler may change in the
+future as well. The `-Z` flags are of course explicitly unstable, but
```

Comment:
there's also the command line flags that are automatically inserted into build products like the the output from `rustc --test` ; see related issue https://github.com/rust-lang/rust/pull/24451